packer/plugin-getter/plugins.go

```diff
@@ -94,6 +93,25 @@ func (rlerr *RateLimitError) Error() string {
 	return s
 }
 
+// PrereleaseError is returned when a getter encounters the install of a pre-release version.
+type PrereleaseError struct {
+	RequestedVersion, ReportedVersion string
+	Source                            string
+}
+
+func (pe *PrereleaseError) Error() string {
+	s := strings.Builder{}
+	s.WriteString("error:\n")
+	s.WriteString("Remote installation of the plugin version " + pe.ReportedVersion + " is unsupported.\n")
+
+	if pe.RequestedVersion != pe.ReportedVersion {
```

Comment:
Yeah in the case of Mondoo. The final version is 10.8.1 but the desc.Version is 10.8.1-dev. In this particular case this is probably an upstream issue. If however, a Plugin does release a 10.8.1-alpha as an actually release then the error message would just indicate that it is not a supported version. This is to not assume its an upstream error. Am I overthinking this?

Reply:
Ah I'm just not clear with what I want to say, I meant to point out that the `if` may be superfluous is all. This error only gets created when the remote plugin version doesn't match the expectation AFAIK at least, but I could be wrong

Reply:
Ah then I'm failing to explain why I added it. If a user tries to install a prerelease directly because the plugin developer tried to release an dev version. Does it make sense for Packer to report that this is likely an upstream issue? 

I'm choosing to display it only if the requested version does not match reported version. If they are both set to prerelease versions then we just don't support I don't know that they need to report anything. Don't want folks reporting issues on plugin repos if they meant to release a prerelease. 

But I feel like I'm over thinking now as I write this.

Reply:
I see gotcha. Then yes since today we're explicitly not supporting installation of pre-release versions, I'd think we won't even reach this error?
But again I haven't tested it yet at all, and AFAIK no-one has made pre-releases for plugins yet, but it may be something to think of.

Reply:
Also: not sure how much we can bulletproof this, but this is likely something we should be checking at callsite even?
Like with `packer init` I know you cannot specify pre-releases in the constraints as constraints explicitly reject pre-releases.
With `packer plugins install` I'm not sure we're checking, but I suspect the implementation is the same, so through a Constraint, which I believe will fail.

Reply:
Yeah I agree. Something to look into afterwards.